### PR TITLE
Fix jaeger metrics path template

### PR DESCRIPTION
**Pull request by douglas-reid**
Fixes: #11962.

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml [hunk 1]
```diff
@@ -23,6 +23,11 @@ spec:
         scheduler.alpha.kubernetes.io/critical-pod: ""
         prometheus.io/scrape: "true"
         prometheus.io/port: "16686"
+{{- if .Values.contextPath }}
+        prometheus.io/path: "{{ .Values.contextPath }}/metrics"
+{{- else }}
+        prometheus.io/path: "/{{ .Values.provider }}/metrics"
+{{- end }}
```
Comment (objectiser) [inline install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml]: LGTM.
Only question is whether `.Values.provider` is required in else. If context path is not set then I believe Jaeger would use endpoint `/metrics`?

Comment (douglas-reid) [reply install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml]: it has to match the `QUERY_BASE_PATH`, which is set to `/{{ .Values.provider }}` later in the template.

Comment (objectiser) [reply install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml]: Ah ok thanks.

**Comment by douglas-reid** [comment]
@duderino any chance of a merge to 1.1 ?

**Comment by duderino** [comment]
Thanks Doug.   Probably 1.1 rc1.  If not that then definitely rc2

**Pull request merged**